File: API/Features/Items/Dtos/ItemReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Items {

    public class ItemReadDto : IBaseEntity, IMetadata {

        // PK
        public int Id { get; set; }
        // Fields
        public string Description { get; set; }
        public byte VatPercent { get; set; }
        public decimal NetPrice { get; set; }
        public decimal GrossPrice { get; set; }
        public bool IsActive { get; set; }
        // Navigation
        public SimpleEntity Color { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/items/classes/dtos/item-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ItemReadDto extends Metadata {

    // PK
    id: number
    // Objects
    color: SimpleEntity
    // Fields
    description: string
    vatPercent: number
    netPrice: number
    grossPrice: number
    isActive: boolean
    // Metadata
    postAt: string
    postUser: string
    putAt: string
    putUser: string

}
